core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DedupAggregator.java

```diff
@@ -198,6 +198,17 @@ private void validateSchema(Schema inputSchema, List<String> uniqueFields,
                              null)
           .withConfigProperty("filterOperation");
       }
+
+      Schema.Type schemaType = field.getSchema().getUnionSchema(0).getType();
+      if ((function.getFunction().equals(DedupConfig.Function.MAX) ||
+              function.getFunction().equals(DedupConfig.Function.MIN)) &&
+              (Schema.Type.STRING.equals(schemaType) || Schema.Type.BOOLEAN.equals(schemaType)
+                      || Schema.Type.BYTES.equals(schemaType))) {
```

Comment:
Please add tests, including decimal (I think it's represented as bytes in physical schema).
Also it's better to have white list here, not black list. We also have arrays and structs

Reply:
yes, agreed. Also tests are failing - need to fix that rn